mne/decoding/time_delaying_ridge.py

```diff
@@ -0,0 +1,186 @@
+            ac[ch0, ch1] = np.concatenate((ac_temp[-len_trf + 1:],
+                                           ac_temp[:len_trf]))
+            if ch0 != ch1:
+                ac[ch1, ch0] = ac[ch0, ch1][::-1]
+
+    # compute the crosscorrelations
+    x_y = np.zeros((n_ch_y, n_ch_x, len_trf))
+    for ch_in in range(n_ch_x):
+        for ch_out in range(n_ch_y):
+            cc_temp = np.fft.irfft(y_fft[ch_out] * X_fft[ch_in].conj(), n_fft)
+            if smin < 0 and smax >= 0:
+                x_y[ch_out, ch_in] = np.append(cc_temp[smin:], cc_temp[:smax])
+            else:
+                x_y[ch_out, ch_in] = cc_temp[smin:smax]
+
+    # make xxt and xy
+    x_xt = _make_x_xt(ac)
+    x_y.shape = (n_ch_y, n_ch_x * len_trf)
+    return x_xt, x_y, n_ch_x
+
+
+def _fit_corrs(x_xt, x_y, n_ch_x, reg_type, alpha, n_ch_in):
+    """Fit the model using correlation matrices."""
+    from scipy import linalg
+    known_types = ('ridge', 'laplacian')
+    if reg_type not in known_types:
+        raise ValueError('reg_type must be one of %s, got %s'
+                         % (known_types, reg_type))
+
+    # do the regularized solving
+    n_ch_out = x_y.shape[0]
+    assert x_y.shape[1] % n_ch_x == 0
+    n_trf = x_y.shape[1] // n_ch_x
+    if reg_type == 'ridge':
+        reg = np.eye(x_xt.shape[0])
+    else:  # if reg_type == 'laplacian':
+        reg = np.eye(n_trf)
+        reg.flat[reg.shape[0] + 1:-reg.shape[0] - 1:reg.shape[0] + 1] = 2
+        reg.flat[1::reg.shape[0] + 1] = -1
+        reg.flat[reg.shape[0]::reg.shape[0] + 1] = -1
+        args = [reg] * n_ch_x
+        reg = linalg.block_diag(*args)
+    mat = x_xt + alpha * reg
+    # From sklearn
+    try:
+        # Note: we must use overwrite_a=False in order to be able to
+        #       use the fall-back solution below in case a LinAlgError
+        #       is raised
+        w = linalg.solve(mat, x_y.T, sym_pos=True, overwrite_a=False)
+    except np.linalg.LinAlgError:
+        warnings.warn('Singular matrix in solving dual problem. Using '
+                      'least-squares solution instead.')
+        w = linalg.lstsq(mat, x_y.T, lapack_driver='gelsy')[0]
+    w = w.T.reshape([n_ch_out, n_ch_in, n_trf])
+    return w
+
+
+def _make_x_xt(ac):
+    len_trf = (ac.shape[2] + 1) // 2
+    n_ch = ac.shape[0]
+    xxt = np.zeros([n_ch * len_trf] * 2)
+    for ch0 in range(n_ch):
+        for ch1 in range(n_ch):
+            xxt_temp = np.zeros((len_trf, len_trf))
+            xxt_temp[0, :] = ac[ch0, ch1, len_trf - 1:]
+            xxt_temp[:, 0] = ac[ch0, ch1, len_trf - 1::-1]
+            for ii in range(1, len_trf):
+                xxt_temp[ii, ii:] = ac[ch0, ch1, len_trf - 1:-ii]
+                xxt_temp[ii:, ii] = ac[ch0, ch1, len_trf - 1:ii - 1:-1]
+            xxt[ch0 * len_trf:(ch0 + 1) * len_trf,
+                ch1 * len_trf:(ch1 + 1) * len_trf] = xxt_temp
+    return xxt
+
+
+class TimeDelayingRidge(BaseEstimator):
```

Comment:
maybe a `See :ref:mne.decoding.ReceptiveField for more information about the point of time delaying etc etc etc`?